### ISSUE TITLE
php artisan optimize - failed when upgrade from 3.6.4 to 4.0-alpha

### ISSUE THREAD

**Issue by nmilanov** [snipe/snipe-it#3514]
#### Expected Behavior (or desired behavior if a feature request)

(what you expect to happen goes here)
Expected to pass this post-install 
-----

#### Actual Behavior

(what actually happens goes here)

Generating autoload files

> Illuminate\Foundation\ComposerScripts::postInstall
> php artisan optimize

  [Symfony\Component\Debug\Exception\FatalThrowableError]
  Class 'Laravel\Tinker\TinkerServiceProvider' not found

**Script php artisan optimize handling the post-install-cmd event returned with error code 1**

-----

#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running - **3.6.4**
- What OS and web server you're running Snipe-IT on - **Ubuntu 16.04.1 LTS**
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) - **manula install**
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error

[2017-04-25 12:35:46] production.ERROR: Symfony\Component\Debug\Exception\FatalThrowableError: Class 'Laravel\Tinker\TinkerServiceProvider' not found in /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php:201
Stack trace:
#0 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(143): Illuminate\Foundation\ProviderRepository->createProvider('Laravel\\Tinker\\...')
#1 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/ProviderRepository.php(60): Illuminate\Foundation\ProviderRepository->compileManifest(Array)
#2 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(543): Illuminate\Foundation\ProviderRepository->load(Array)
#3 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/RegisterProviders.php(17): Illuminate\Foundation\Application->registerConfiguredProviders()
#4 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(208): Illuminate\Foundation\Bootstrap\RegisterProviders->bootstrap(Object(Illuminate\Foundation\Application))
#5 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(270): Illuminate\Foundation\Application->bootstrapWith(Array)
#6 /var/www/snipeit/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(115): Illuminate\Foundation\Console\Kernel->bootstrap()
#7 /var/www/snipeit/artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 {main}

- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error

> Illuminate\Foundation\ComposerScripts::postInstall
> php artisan optimize

- If a stacktrace is provided in the error, include that too.

![error](https://cloud.githubusercontent.com/assets/16882507/25379580/2b422cfc-29b6-11e7-874c-accb5b624db8.png)

- Any errors that appear in your browser's error console.
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
- Include what you've done so far in the installation, and if you got any error messages along the way.

1. Download alpha version from Github
2. Copy .env file from previous version
3. Install dependencies 
cd install-dir
curl -sS https://getcomposer.org/installer | php
4. php composer.phar install --no-dev --prefer-source

- Indicate whether or not you've manually edited any data directly in the database

No

**Comment by snipe** [comment]
What was the composer command you ran?

**Comment by snipe** [comment]
Ah, nevermind, sorry, I see it. I think that's because you ran it as --no-dev (which you were supposed to). I'll push out an updated composer lock in a moment.

**Comment by snipe** [comment]
If you grab the latest off develop, this fix should be in there. Otherwise you can run `php composer.phar install --prefer-source` instead of `php composer.phar install --no-dev --prefer-source` and it should work with what you have.

**Comment by nmilanov** [comment]
When I try run

> php composer.phar install --prefer-source

I get this message:

Generating autoload files
> Illuminate\Foundation\ComposerScripts::postInstall
> php artisan optimize


  [Symfony\Component\HttpKernel\Exception\NotFoundHttpException]


**Script php artisan optimize handling the post-install-cmd event returned with error code 1**

I can't see anything in laravel.log
From browser get error 404.

**Comment by msil** [comment]
I'm having the same issue as above when attempting to either upgrade 3.6.4 to 4.0-alpha or perform a fresh install of 4.0-alpha

Is this a routing issue?

**Comment by nmilanov** [comment]
@snipe - should I open new issue with this error?

**Comment by snipe** [comment]
@nmilanov No need, I've reopened this one. We're having a helluva time reproducing this. Can you try running `php artisan route:clear`?

**Comment by nmilanov** [comment]
Error is the same:

 [Symfony\Component\HttpKernel\Exception\NotFoundHttpException]

**Comment by snipe** [comment]
When you load the actual web page, what does it show?

**Comment by nmilanov** [comment]
404 Page not found.

Sad panda. We could not find the page you were looking for. You should maybe return to the dashboard.

**Comment by snipe** [comment]
Try deleting all of the files inside the `storage/framework` directory

**Comment by snipe** [comment]
(Also let me know if `php artisan route:list ` gives you anything useful)

**Comment by snipe** [comment]
(And also `composer dump-autoload`)

**Comment by nmilanov** [comment]
1. there were only 2 php files in ./storage/framework/views - deleted
2. `php artisan route:list` - [Symfony\Component\HttpKernel\Exception\NotFoundHttpException]
3. `composer dump-autoload` - **Generating autoload files** (with no errors)

**Comment by snipe** [comment]
And after those three things, the website is still 404?

**Comment by nmilanov** [comment]
yep - still **404 Page not found**
I think I start all over, delete all, download again last alpha and try again (will keep only .env file and the database)

**Comment by snipe** [comment]
Also try removing /bootstrap/compiled.php and re-running composer

**Comment by snipe** [comment]
Also try deleting the bootstrap/cache/compiled.php file and re-running composer

**Comment by snipe** [comment]
Another thing you can try is in composer.json:

change:

```
"scripts": {
        "post-root-package-install": [
            "php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "php artisan key:generate"
        ],
        "post-install-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postInstall",
            "php artisan optimize"
        ],
        "post-update-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postUpdate",
            "php artisan optimize"
        ]
    },
```

to:

```
"scripts": {
        "post-root-package-install": [
            "php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "php artisan key:generate"
        ]
    },
```

**Comment by nmilanov** [comment]
Still no success.
I’ll start from the beginning and I’ll write down all I do and let you know the result.
Thanks for your help.

**Comment by snipe** [comment]
Gaaah. I'm so sorry this is proving so difficult, but I appreciate your help in getting this sorted. :(

**Comment by snipe** [comment]
(If you're not already doing it, it's probably best to test the alpha just by using the develop branch versus downloading the zip, that way you can pick up whatever changes I push through.)

**Comment by nmilanov** [comment]
I get the latest version alpha-4.0
`git clone -b v4.0-alpha https://github.com/snipe/snipe-it snipeit`

then try to install dependencies:
`curl -sS https://getcomposer.org/installer | php`
and all looks good, then 
`php composer.phar install --no-dev --prefer-source`
> Loading composer repositories with package information
> Installing dependencies from lock file
> .....
> Generating autoload files
> Illuminate\Foundation\ComposerScripts::postInstall
> php artisan optimize


  [Symfony\Component\Debug\Exception\FatalThrowableError]
  Class 'Laravel\Tinker\TinkerServiceProvider' not found


**Script php artisan optimize handling the post-install-cmd event returned with error code 1**

Should I try w/o `--no-dev` ?

**Comment by snipe** [comment]
@nmilanov as I mentioned, you'll want to pull the latest from develop, not from the tagged release

**Comment by snipe** [comment]
I've been able to sort of reproduce this, and I patched the composer file to remove the problematic post-commands. (I don't know why they were failing, but Google isn't giving me any love there.)

Now I'm getting a 404 on the homepage and on any artisan commands, and am trying to figure out what's going on there. So the good news is that I can actually reproduce it. The bad news is I don't know why it's broken.
